.github/workflows/reusable-docker.yml

```diff
@@ -0,0 +1,141 @@
+name: Build, tag and push docker image to ghcr.io
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+on:
+  workflow_call:
+    inputs:
+      environment:
+        required: true
+        type: string
+      package_name:
+        required: true
+        type: string
+      project_path:
+        required: true
+        type: string
+      depot_project_id:
+        required: true
+        type: string
+      test_port:
+        required: false
+        default: '1341'
+        type: string
+      health_check:
+        required: false
+        default: false
+        type: boolean
+      # local tag is used to tag the image locally, you can find it in package.json
+      local_tag:
+        required: true
+        type: string
+      # supported tags: dev, stg, prod
+      env_tag:
+        required: false
+        type: string
```

Comment:
`environment` - the GH secrets environment
`package_name` - the name under which the docker image will be published on GH Packages
`project_path` - the path to the project in the monorepo
`depot_project_id` - the depot project id, currently we build everything under one project (it requires to set up the OIDC for each project and I had some issues with the Depot)
`test_port` - you can test the image after it's built in test mode and by doing a health-check
`health_check` - whether to perform a health-check
`local_tag` - the tag under which the image is built, it should align with the name in the project command `docker:build:depot`
`env_tag` - the name of the tag, could be dev, stg, prod